packages/@aws-cdk/aws-apigateway/lib/cors.ts

```diff
@@ -0,0 +1,105 @@
+import { Duration } from '@aws-cdk/core';
+import { ALL_METHODS } from './util';
+
+export interface CorsOptions {
+  /**
+   * Specifies the response status code returned from the OPTIONS method.
+   *
+   * @default 204
+   */
+  readonly statusCode?: number;
+
+  /**
+   * The Access-Control-Allow-Origin response header indicates whether the
+   * response can be shared with requesting code from the given origin.
+   *
+   * Specifies the list of origins that are allowed to make requests to this resource.
+   *
+   * @see https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Access-Control-Allow-Origin
+   */
+  readonly allowOrigins: string[];
+
+  /**
+   * The Access-Control-Allow-Headers response header is used in response to a
+   * preflight request which includes the Access-Control-Request-Headers to
+   * indicate which HTTP headers can be used during the actual request.
+   *
+   * @see https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Access-Control-Allow-Headers
+   * @default Cors.DEFAULT_HEADERS
+   */
+  readonly allowHeaders?: string[];
+
+  /**
+   * The Access-Control-Allow-Methods response header specifies the method or
+   * methods allowed when accessing the resource in response to a preflight request.
+   *
+   * If `ANY` is specified, it will be expanded to `Cors.ALL_METHODS`.
+   *
+   * @see https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Access-Control-Allow-Methods
+   * @default Cors.ALL_METHODS
+   */
+  readonly allowMethods?: string[];
+
+  /**
+   * The Access-Control-Allow-Credentials response header tells browsers whether
+   * to expose the response to frontend JavaScript code when the request's
+   * credentials mode (Request.credentials) is "include".
+   *
+   * When a request's credentials mode (Request.credentials) is "include",
+   * browsers will only expose the response to frontend JavaScript code if the
+   * Access-Control-Allow-Credentials value is true.
+   *
+   * Credentials are cookies, authorization headers or TLS client certificates.
+   * @see https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Access-Control-Allow-Credentials
+   * @default false
+   */
+  readonly allowCredentials?: boolean;
+
+  /**
+   * The Access-Control-Max-Age response header indicates how long the results of
+   * a preflight request (that is the information contained in the
+   * Access-Control-Allow-Methods and Access-Control-Allow-Headers headers)
+   * can be cached.
+   *
+   * To disable caching altogther use `disableCache: true`.
```

Comment:
Wouldn't specifying `0` (or less) here be an effective way to signal "I don't want no caching"? I'm not too sure the extra flag is actually so useful.

Reply:
The [spec](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Access-Control-Max-Age) doesn't say anything about `0` and we can't express `-1` as a `Duration`.